### DISCUSSION TITLE
Trivy scan raises a warning message when analyse Yarn alias

### DISCUSSION THREAD

**Discussion by Sharpforce** [aquasecurity/trivy#5814] [Bugs]
### Description

Trivy issues a warning message when analyzing a package.json file that uses a Yarn alias : npm:foo@1.0.1.

### Desired Behavior

It is expected that Trivy does not raise any warnings but manages to parse the input correctly. In the specified case, the latest version should be considered, but it is also possible to specify the version, for example: "npm:@types/jsonstream@^0.8.33".

### Actual Behavior

When Trivy scans my Yarn project (with the presence of yarn.lock and package.json), the following warning message is displayed: 'WARN Unable to parse "package.json" to remove dev dependencies: unable to walk dependencies: unable to match version for @types/JSONStream.'.

The affected line is the following. It utilizes the concept of Yarn's custom alias (https://classic.yarnpkg.com/lang/en/docs/cli/add/#toc-yarn-add-alias) :

"devDependencies": {
  "@types/JSONStream": "npm:@types/jsonstream"
}

The reasons for this usage, specifically for this dependency, stem from a known character case error, the resolution of which involves the use of a custom alias (see the last comment on https://github.com/DefinitelyTyped/DefinitelyTyped/issues/9716).

### Reproduction Steps

```bash
Step 1 : create the following package.json file :

{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/JSONStream": "npm:@types/jsonstream"
  }
}

Step 2 : Execute yarn install
Step 3 : trivy scan
```

### Target

Filesystem

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Debug Output

```bash
DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
DEBUG	Ignore statuses	{"statuses": null}
DEBUG	cache dir:  .trivycache
DEBUG	DB update was skipped because the local DB is the latest
DEBUG	DB Schema: 2, UpdatedA
```

### Operating System

Linux

### Version

```bash
0.47
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @Sharpforce 
Thanks for your report!

I created #5817 for this task.

Regards, Dmitriy
